packages/node-utils/src/edge-to-node/handler.ts

```diff
@@ -0,0 +1,52 @@
+import type { IncomingMessage, ServerResponse } from 'node:http'
+import type {
+  WebHandler,
+  NodeHandler,
+  BuildDependencies,
+  RequestOptions,
+} from '../types'
+import { buildToRequest } from '../node-to-edge/request'
+import { mergeIntoServerResponse, toOutgoingHeaders } from './headers'
+import { toToReadable } from './stream'
+
+export function buildToNodeHandler(
+  dependencies: BuildDependencies,
+  options: RequestOptions
+) {
+  const toRequest = buildToRequest(dependencies)
+  return function toNodeHandler(webHandler: WebHandler): NodeHandler {
+    return (request: IncomingMessage, response: ServerResponse) => {
+      const maybePromise = webHandler(toRequest(request, options))
+      if (maybePromise instanceof Promise) {
+        maybePromise.then((webResponse) =>
+          mergeToServerResponse(webResponse, response)
+        )
+      } else {
+        mergeToServerResponse(maybePromise, response)
+      }
+    }
+  }
+}
+
+function mergeToServerResponse(
```

Comment:
**suggestion:** to be consistent with `mergeIntoServerResponse`

```suggestion
function mergeIntoServerResponse(
```

Reply:
hum, actually I messed this up. `mergeIntoServerResponse()` is an utility function coming from `./header.ts` file, only operating on headers. This internal function should be named completely differently.

Reply:
Oh didn't notice it's a private function. Then LGTM 👍